commons/src/main/java/io/aiven/kafka/connect/common/grouper/TopicPartitionRecordGrouper.java

```diff
@@ -50,14 +52,14 @@ class TopicPartitionRecordGrouper implements RecordGrouper {
             DateTimeFormatter.ofPattern("dd"), "HH", DateTimeFormatter.ofPattern("HH"));
 
     private final Template filenameTemplate;
+    // Offsets are a Long and Partitions are an Integer
+    private final Map<TopicPartition, Pair<Long, Integer>> currentHeadRecords = new HashMap<>();
```

Comment:
Can we go the same way as in Kafka and similar to `TopicPartition` create something like `PartitionOffset` or similar? Otherwise all this `gteLeft` and `getRight` calls a little bit hard to follow

Reply:
Yes, that is a good shout to improve readability will happily make that change

Reply:
Updated thank you!